pymc4/distributions/plate.py

```diff
@@ -0,0 +1,204 @@
+"""The Plate distribution class."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+# Dependency imports
+import numpy as np
+
+# import tensorflow.compat.v2 as tf
+import tensorflow as tf
+
+from tensorflow_probability.python.distributions import distribution as distribution_lib
+from tensorflow_probability.python.distributions import kullback_leibler
+from tensorflow_probability.python.internal import assert_util
+from tensorflow_probability.python.internal import distribution_util
+from tensorflow_probability.python.internal import prefer_static
+from tensorflow_probability.python.internal import tensorshape_util
+
+
+def _make_summary_statistic(attr):
+    """Factory for implementing summary statistics, eg, mean, stddev, mode."""
+
+    def _fn(self, **kwargs):
+        """Implements summary statistic, eg, mean, stddev, mode."""
+        x = getattr(self.distribution, attr)(**kwargs)
+        shape = prefer_static.concat(
+            [
+                prefer_static.ones(
+                    prefer_static.rank_from_shape(self.plate_shape), dtype=self.plate_shape.dtype
+                ),
+                self.distribution.batch_shape_tensor(),
+                self.distribution.event_shape_tensor(),
+            ],
+            axis=0,
+        )
+        x = tf.reshape(x, shape=shape)
+        shape = prefer_static.concat(
+            [
+                self.plate_shape,
+                self.distribution.batch_shape_tensor(),
+                self.distribution.event_shape_tensor(),
+            ],
+            axis=0,
+        )
+        return tf.broadcast_to(x, shape)
+
+    return _fn
+
+
+class Plate(distribution_lib.Distribution):
```

Comment:
As a design choice - what do you think in terms of editing the `tfd.Sample` to also accept plate_shape? In that case we can also allow user to do `tfd.Sample(tfd.Plate(tfd.Somedistribution, plate_shape), sample_shape)` in one pass.

Reply:
Oh, I hadn't thought of editing `tfd.Sample`. Maybe we could do that and we would be able to drop the boolean flag. I'm not completely sure what's best here

Reply:
Not sure as well, @jvdillon?